### PR TITLE
Settings page

### PR DESCRIPTION
**Pull request by jowczar**
Adds Firebase storage, Firebase user updates, usage of form components, new package for validation: Yup, and plenty of other stuff that we will use in Admin Panel later

### CODE REVIEW

File: frontend/src/app/(authorized)/settings/form.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    UserData & { password: string; currentPassword: string }
+  >({
+    mode: "onTouched",
+    resolver: yupResolver(formSchema),
+    defaultValues: {
+      email,
+      displayName,
+    },
+  });
+  const watchPassword = watch("password");
```
Comment (ShanghaiG) [inline frontend/src/app/(authorized)/settings/form.tsx:40]: how is this watch working?

File: frontend/package.json [hunk 2]
```diff
@@ -30,7 +31,8 @@
     "react-responsive-carousel": "^3.2.23",
     "react-toastify": "^9.1.3",
     "typescript": "5.0.4",
-    "uuid": "^9.0.0"
+    "uuid": "^9.0.0",
```
Comment (ShanghaiG) [inline frontend/package.json:34]: tbh, do we really need uuid? we could use already implemented in js function, `crypto.randomUUID()`

**Review by ShanghaiG — APPROVED** [review]

File: frontend/package.json (hunk 2 above)
Comment (jowczar) [reply frontend/package.json:34]: @ShanghaiG good catch, I will remove it

File: frontend/src/app/(authorized)/settings/form.tsx (hunk 1 above)
Comment (jowczar) [reply frontend/src/app/(authorized)/settings/form.tsx:40]: @ShanghaiG it's a method from react form hooks
https://react-hook-form.com/docs/useform/watch

it's a subscriber that allows you to access present values in the form, and also look for changes in them

**Pull request merged**